include/path.hpp

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <eigen3/Eigen/Dense>
+#include <eigen3/Eigen/Core>
+#include <glm/glm.hpp>
+
+
+// -------------------------------------------------------------------------------------------------------------------//
+
+// Misc definitions
+
+
+// Flags for boundary conditions
+// Since Eigen arrays don't work well with enum types, we just take integer flags
+typedef int bc_t;
+static const bc_t BC_T_START = 0, BC_ZERO = 0, BC_NATURAL = 1, BC_CLAMP = 2, BC_T_END = 3;
```

Comment:
No, actually it still goes wrong with stuff like this:
```C++
error: no matching function for call to ‘log(Eigen::internal::significant_decimals_default_impl<boundary_condition, false>::RealScalar)’
     return cast<RealScalar,int>(ceil(-log(NumTraits<RealScalar>::epsilon())/log(RealScalar(10))));
```
Probably there's a way to tell Eigen how to deal with this type.